solr/core/src/test/org/apache/solr/cloud/hdfs/HdfsTestUtil.java

```diff
@@ -239,4 +253,24 @@ public static String getURI(MiniDFSCluster dfsCluster) {
     }
   }
 
+  /**
+   * By default in JDK9+, the ForkJoinWorkerThreadFactory does not give SecurityManager permissions
+   * to threads that are created. This works around that with a custom thread factory.
+   * See SOLR-9515 and HDFS-14251
+   * Used in org.apache.hadoop.hdfs.server.datanode.fsdataset.impl.BlockPoolSlice
+   */
+  public static class HDFSForkJoinThreadFactory implements ForkJoinPool.ForkJoinWorkerThreadFactory {
+    @Override
+    public ForkJoinWorkerThread newThread(ForkJoinPool pool) {
+      ForkJoinWorkerThread worker = new SecurityManagerWorkerThread(pool);
+      worker.setName("solr-hdfs-threadpool-" + worker.getPoolIndex());
```

Comment:
Name used in BadHdfsThreadFilter since this pool is not shutdown correctly by BlockPoolSlice and not easily worked around. See SOLR-9515 and HDFS-14251